### PR TITLE
Refactor: address remaining issues found by Spearbit audit

### PR DESCRIPTION
**Pull request by amusingaxl**
- docs(StakingRewards): add warning regarding `notifyRewardAmount` usage
- refactor(StakingRewards): add check in constructor

### CODE REVIEW

**Comment by oldchili** [comment]
Should this PR also revert https://github.com/makerdao/endgame-toolkit/commit/8ff5f3935eab27ebbadf54cbe9057facd5704f3c ?

**Comment by amusingaxl** [comment]
> Should this PR also revert [8ff5f39](https://github.com/makerdao/endgame-toolkit/commit/8ff5f3935eab27ebbadf54cbe9057facd5704f3c) ?

I don't believe so. We usually want to avoid failing scripts. At the very least, it prevents the deployer from wasting some gas in case of a problem in the config.
The script is also some kind of documentation, even though in this case it is pretty obvious from the code, but that is not always the case.

File: README.md
Comment (oldchili) [inline README.md]: Up to you, but I'd just remove the diff links from everywhere and would just leave the permalink to the original code (in one place). Anyone can do the diff given the permalink. 
IMO the code should be easily maintainable, which means not having to look for many places to update the documentation on each change.

File: src/synthetix/StakingRewards.sol
Comment (oldchili) [inline src/synthetix/StakingRewards.sol]: Otherwhise => Otherwise

Comment (oldchili) [inline src/synthetix/StakingRewards.sol]: Neat - I wouldn't put the @dev natspec tag, as natspec is not used in other places in this contract.

**Review by sunbreak1211 — COMMENTED** [review]
PR looks good to me, I'll wait for the typo fix and response in the other comment to give my approval.

**Comment by telome** [comment]
LGTM, will approve once the comments above are addressed.

File: README.md
Comment (amusingaxl) [reply README.md]: Done.

File: src/synthetix/StakingRewards.sol
Comment (amusingaxl) [reply src/synthetix/StakingRewards.sol]: Fixed.

Comment (amusingaxl) [reply src/synthetix/StakingRewards.sol]: Updated.

File: README.md
Comment (oldchili) [reply README.md]: I meant removing also from StakingRewards.sol.

**Review by oldchili — APPROVED** [review]
LGTM

**Review by sunbreak1211 — APPROVED** [review]
LGTM

**Review by telome — APPROVED** [review]
LGTM

**Pull request merged**
